db/view/view.cc

```diff
@@ -2064,11 +2121,19 @@ future<> view_update_generator::mutate_MV(
     co_await max_concurrent_for_each(view_updates, max_concurrent_updates, [&] (frozen_mutation_and_schema mut) mutable -> future<> {
         auto view_token = dht::get_token(*mut.s, mut.fm.key());
         auto view_ermp = erms.at(mut.s->id());
-        auto target_endpoint = get_view_natural_endpoint(me, base_ermp, view_ermp, replication, base_token, view_token,
+        auto [target_endpoint, no_pairing_endpoint] = get_view_natural_endpoint(me, base_ermp, view_ermp, replication, base_token, view_token,
                 use_legacy_self_pairing, use_tablets_rack_aware_view_pairing, cf_stats);
         auto remote_endpoints = view_ermp->get_pending_replicas(view_token);
         auto sem_units = seastar::make_lw_shared<db::timeout_semaphore_units>(pending_view_updates.split(memory_usage_of(mut)));
-
+        if (no_pairing_endpoint) {
+            // The no_pairing_endpoint can appear during a replication factor increase when the base tablet migration finished after
+            // the corresponding (for current token) view tablet migration. In this case, the view replica is no longer
+            // in migration (so it's not present in the pending endpoints list), but the base replica still is. We don't pair pending
```

Comment:
After RF 0->N changes the base replica won't necessarily be in migration - it will start the migration eventually.